### ISSUE TITLE
Lack of input checks lead to possibility of user error

### ISSUE THREAD

**Issue by jarolrod** [bitcoin-core/gui#198]
### Wallet Creation Issues
1. **Can enter unlimited amount of characters**
    - The wallet creation dialogue allows me to input an unlimited amount of characters. A long enough filename will lead to a crash with error: `boost::filesystem::status: File name too long`
    - Instead of crashing, it would be nice to know that I exceeded the file length limit, and change my new wallet name
2. **No consideration of POSIX portability**
    - When creating a wallet, one is allowed to enter any character. This means that I can make a wallet on one system that is not portable to another system

### Receive Label and Message Issues
When creating a new payment request under the `Receive` tab, you are allowed to input an unlimited amount of characters in both the `Label` and `Message` fields. This will lead to an unusable URI with no QR code generated. While this would be a rare occurrence, this should not be allowed to happen. A user should not be allowed to create a URI that is unusable. It also creates a ridiculously long window, see below:

<img width="3284" alt="Screen Shot 2021-01-23 at 10 15 33 AM" src="https://user-images.githubusercontent.com/23396902/105606568-2ee9fa80-5d68-11eb-9762-beec8b2ee1ab.png">

**Comment by MarcoFalke** [comment]
>     When creating a wallet, one is allowed to enter any character. This means that I can make a wallet on one system that is not portable to another system

You can simply rename the wallet folder while moving it to the other system, no?

**Comment by jarolrod** [comment]
@MarcoFalke Yeah, that is what you would have to do. I guess it comes down to should we allow behavior that will lead to this?
